torchchat/model.py

```diff
@@ -442,8 +442,11 @@ def setup_caches(self, max_batch_size, max_seq_length):
         self.max_seq_length = max_seq_length
         self.max_batch_size = max_batch_size
         for b in self.layers.values():
-            b.attention.kv_cache = KVCache(
-                max_batch_size, max_seq_length, self.config.n_local_heads, head_dim
```

Comment:
This is fine, just calling it out as a note to self

Local `head_dim` is defined as `self.config.dim // self.config.n_heads`. 

In Attention, head_dim is pulled from `config.head_dim`. Recall that TransformerArgs (config) is also initialized as `self.config.dim // self.config.n_heads`